internal/cli/uninstall.go

```diff
@@ -55,24 +56,44 @@ func (c *UninstallCommand) Run(args []string) int {
 		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
 		return 1
 	}
-	c.ui.Output(
-		"Uninstalling Waypoint server with context %q",
-		contextDefault,
-		terminal.WithSuccessStyle(),
-	)
 
-	// Get the platform early so we can validate it.
-	p, ok := serverinstall.Platforms[strings.ToLower(c.platform)]
-	if !ok {
-		if c.platform == "" {
+	var ctxConfig *clicontext.Config
+	if contextDefault != "" {
+		ctxConfig, err = c.contextStorage.Load(contextDefault)
+		if err != nil {
+			c.ui.Output("Error loading context %q: %s", contextDefault, err.Error(), terminal.WithErrorStyle())
+			return 1
+		}
+	}
+
+	serverPlatform := c.platform
+	if ctxConfig.Server.Platform == "" && serverPlatform != "" {
+		c.ui.Output(
+			"No platform set on server context. Will attempt to uninstall requested "+
+				"platform %q",
+			serverPlatform,
+			terminal.WithWarningStyle(),
+		)
+	} else if serverPlatform == "" {
+		// attempt to set the server platform so the platform flag isn't required.
+		serverPlatform = ctxConfig.Server.Platform
+
+		if serverPlatform == "" {
+			// It's still empty
 			c.ui.Output(
-				"The -platform flag is required.",
+				"Cannot determine what platform to uninstall Waypoint. "+
+					"The -platform flag is required since the server context did not include "+
+					"a server platform.",
 				terminal.WithErrorStyle(),
 			)
 
 			return 1
 		}
+	}
```

Comment:
Should we reject if there is a context and a serverPlatform and they conflict about the Platform value?

Reply:
Sounds like a good idea to me!